File: tradex-middleware/src/service/ReportService.js
// service/ReportService.js
const axios = require('axios');

const BACKEND_URL = 'http://spring-app:8081/reports';

// ----------------- Portfolio report -----------------
exports.getPortfolioReport = async (clientId) => {
    console.log('Service: Getting portfolio report for clientId:', clientId);
    const response = await axios.get(`${BACKEND_URL}/portfolio/${clientId}`);
    console.log('Service: Portfolio report response:', response.data);
    return response.data;
};

// ----------------- Trade history report -----------------
exports.getTradeHistoryReport = async (clientId) => {
    console.log('Service: Getting trade history report for clientId:', clientId);
    const response = await axios.get(`${BACKEND_URL}/trade-history/${clientId}`);
    console.log('Service: Trade history report response:', response.data);
    return response.data;
};

File: tradex-service/src/main/java/com/example/tradex_service/Service/ReportService.java
package com.example.tradex_service.Service;

import com.example.tradex_service.DTO.Response.PortfolioReportDto;
import com.example.tradex_service.DTO.Response.TradeHistoryReportDto;

public interface ReportService {
    PortfolioReportDto getPortfolioReport(String clientId);
    TradeHistoryReportDto getTradeHistoryReport(String clientId);
}